geowebcache/sqlite/src/main/java/org/geowebcache/sqlite/MbtilesBlobStore.java

```diff
@@ -0,0 +1,538 @@
+/**
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Nuno Oliveira, GeoSolutions S.A.S., Copyright 2016
+ */
+package org.geowebcache.sqlite;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geotools.mbtiles.GeoToolsMbtilesUtils;
+import org.geotools.mbtiles.MBTilesFile;
+import org.geotools.mbtiles.MBTilesMetadata;
+import org.geotools.mbtiles.MBTilesTile;
+import org.geotools.sql.SqlUtil;
+import org.geowebcache.storage.BlobStoreListener;
+import org.geowebcache.storage.BlobStoreListenerList;
+import org.geowebcache.storage.StorageException;
+import org.geowebcache.storage.TileObject;
+import org.geowebcache.storage.TileRange;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Blobstore that store the tiles in a sqlite database using the mbtiles specification.
+ */
+public final class MbtilesBlobStore extends SqliteBlobStore {
+
+    private static Log LOGGER = LogFactory.getLog(MbtilesBlobStore.class);
+
+    // pattern for matching the name of a file that contains mbtiles metadata (layerName.metadata)
+    private final static Pattern MBTILES_METADATA_FILE_NAME_PATTERN = Pattern.compile("(.*?)\\.metadata");
+
+    // sqlite database that will contain layers metadata
+    private final File metadataFile;
+
+    // if true we will prefer delete a file rather than delete only a few tiles (avoiding vacuum)
+    private final boolean eagerDelete;
+
+    // if false we will not care about tile create time (expiration rules will not work)
+    private final boolean useCreateTime;
+
+    private final BlobStoreListenerList listeners;
+
+    // parsed mbtiles metadata indexed per layer
+    private final Map<String, MBTilesMetadata> layersMetadata = new ConcurrentHashMap<>();
+
+    // Executor that can be used to perform parallel operations
+    private final ExecutorService executorService;
```

Comment:
The executor service should be shut down on application undeploy (again, you can listen to spring events to do that).